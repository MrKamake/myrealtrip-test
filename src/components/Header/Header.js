import React, { Component } from 'react';
import './Header.scss';
import { Link } from 'react-router-dom';
import Main from '../../images/main-logo.png';

class Header extends Component {
  handleClick = selection => {
    const { updateProductList } = this.props;
    updateProductList(selection);
  };

  render() {
    const { sortMessage, selectedListMessage } = this.props;

    return (
      <div className="Header">
        <Link to="/main" className="main-logo">
          <img src={Main} alt="Main Logo" />
        </Link>
        <div className="buttons-wrapper">
          <div className="sort-buttons-wrapper">
            <Link to="/sorted-price/high">
              <button
                className={`high-price-button ${
                  sortMessage === 'high' ? 'highlight' : ''
                }`}
                onClick={() => this.handleClick('high')}
              >
                가격 높은순
              </button>
            </Link>
            <Link to="/sorted-price/low">
              <button
                className={`low-price-button ${
                  sortMessage === 'low' ? 'highlight' : ''
                }`}
                onClick={() => this.handleClick('low')}
              >
                가격 낮은순
              </button>
            </Link>
          </div>
          <div className="list-buttons-wrapper">
            <Link to="/products/wish">
              <button
                className={`wish-list-button ${
                  selectedListMessage === 'wish' ? 'highlight' : ''
                }`}
                onClick={() => this.handleClick('wish')}
              >
                위시 리스트
              </button>
            </Link>
            <Link to="/main">
              <button
                className={`product-list-button ${
                  selectedListMessage === 'default' ? 'highlight' : ''
                }`}
                onClick={() => this.handleClick('default')}
              >
                상품 리스트
              </button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Header;
