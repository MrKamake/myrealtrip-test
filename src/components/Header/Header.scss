@import '../../styles/utils.scss';

.Header {
  position: fixed;
  display: flex;
  z-index: 1;
  width: 100%;
  height: 73px;
  background-color: $mrtblue;
  .main-logo {
    cursor: pointer;
    padding: 20px 0 0 15%;
    img {
      width: 128px;
      height: 28px;
    }
  }
  .buttons-wrapper {
    display: inline-flex;
    justify-content: flex-end;
    padding: 20px 14% 0 0;
    width: 64%;
    height: 100%;
    button {
      cursor: pointer;
      padding: 6px 12px;
      height: 36px;
      border-radius: 3px;
      font-size: 15px;
      font-weight: 600;
      color: white;
      background-color: $transparent;
    }
    a {
      height: 36px;
      border-radius: 3px;
    }
    .highlight {
      font-weight: bold;
      color: #2a96ed;
      background-color: $lightwhite;
    }
    div {
      display: inline-flex;
      justify-content: space-around;
      width: 200px;
    }
  }
}
