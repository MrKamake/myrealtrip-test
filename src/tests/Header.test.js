import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../components/Header/Header';

configure({ adapter: new Adapter() });

describe('<Header />', () => {
  let wrapper;
  let sortMessage;
  let updateProductList;
  let selectedListMessage;

  beforeEach(() => {
    sortMessage = '';
    updateProductList = jest.fn();
    selectedListMessage = '';

    wrapper = shallow(
      <Header
        sortMessage={sortMessage}
        updateProductList={updateProductList}
        selectedListMessage={selectedListMessage}
      />
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('should contain clickable buttons', () => {
    it('on click high price button', () => {
      const highPriceButton = wrapper.find('.high-price-button');
      expect(highPriceButton.length).toBe(1);
      highPriceButton.simulate('click');
      expect(updateProductList.mock.calls.length).toEqual(1);
    });
    it('on click low price button', () => {
      const lowPriceButton = wrapper.find('.low-price-button');
      expect(lowPriceButton.length).toBe(1);
      lowPriceButton.simulate('click');
      expect(updateProductList.mock.calls.length).toEqual(1);
    });

    it('on click wish list button', () => {
      const wishListButton = wrapper.find('.wish-list-button');
      expect(wishListButton.length).toBe(1);
      wishListButton.simulate('click');
      expect(updateProductList.mock.calls.length).toEqual(1);
    });

    it('on click product list button', () => {
      const defaultListButton = wrapper.find('.product-list-button');
      expect(defaultListButton.length).toBe(1);
      defaultListButton.simulate('click');
      expect(updateProductList.mock.calls.length).toEqual(1);
    });
  });
});
