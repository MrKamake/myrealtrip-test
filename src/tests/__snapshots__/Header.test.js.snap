// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div
  className="Header"
>
  <Link
    className="main-logo"
    to="/main"
  >
    <img
      alt="Main Logo"
      onClick={[Function]}
      src="main-logo.png"
    />
  </Link>
  <div
    className="buttons-wrapper"
  >
    <div
      className="sort-buttons-wrapper"
    >
      <Link
        to="/sorted-price/high"
      >
        <button
          className="high-price-button "
          onClick={[Function]}
        >
          가격 높은순
        </button>
      </Link>
      <Link
        to="/sorted-price/low"
      >
        <button
          className="low-price-button "
          onClick={[Function]}
        >
          가격 낮은순
        </button>
      </Link>
    </div>
    <div
      className="list-buttons-wrapper"
    >
      <Link
        to="/products/wish"
      >
        <button
          className="wish-list-button "
          onClick={[Function]}
        >
          위시 리스트
        </button>
      </Link>
      <Link
        to="/main"
      >
        <button
          className="product-list-button "
          onClick={[Function]}
        >
          상품 리스트
        </button>
      </Link>
    </div>
  </div>
</div>
`;
